hs_core/views/utils.py

```diff
@@ -57,18 +58,22 @@ def authorize(request, res_id, edit=False, view=False, full=False, superuser=Fal
     except ObjectDoesNotExist:
         raise NotFound(detail="No resource was found for resource id:%s" % res_id)
 
-    if user.is_authenticated():
+    if discoverable and (edit is False and view is False and full is False and superuser is False):
+        authorized = res.raccess.discoverable
+    elif user.is_authenticated():
         has_edit = user.uaccess.can_change_resource(res)
         has_view = user.uaccess.can_view_resource(res)
         has_full = user.uaccess.owns_resource(res)
 
-        authorized = (edit and has_edit) or \
+        authorized = (discoverable and res.raccess.discoverable) or \
+                     (edit and has_edit) or \
                      (view and has_view) or \
                      (full and has_full) or \
                      (superuser and (user.uaccess.admin or user.is_superuser))
     else:
         has_view = res.raccess.public
-        authorized = view and has_view
+        authorized = (discoverable and res.raccess.discoverable) or \
+                     (view and has_view)
```

Comment:
I think if has_view is set to res.raccess.public or res_raccess.discoverable, we don't need the extra check for discoverable here.